zebra-chain/src/commitments/sprout.rs

```diff
@@ -0,0 +1,57 @@
+//! Sprout commitment types.
+
+#![allow(clippy::unit_arg)]
+
+use std::io;
+
+use sha2::{Digest, Sha256};
+
+use crate::{
+    notes::sprout::Note,
+    serialization::{ReadZcashExt, SerializationError, ZcashDeserialize, ZcashSerialize},
+};
+
+/// The randomness used in the Pedersen Hash for note commitment.
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct CommitmentRandomness(pub [u8; 32]);
+
+impl AsRef<[u8]> for CommitmentRandomness {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+/// Note commitments for the output notes.
+#[derive(Clone, Copy, Debug)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct NoteCommitment(pub(crate) [u8; 32]);
```

Comment:
crossreferencing with https://github.com/ZcashFoundation/zebra/pull/596#discussion_r465944775

The `sprout::NoteCommitment` here makes the opposite choice as the `sapling::NoteCommitment`.  A priori it seems like situation is basically similar in both cases, so it would seem like whichever choice we make, it should be consistent between the two cases.

Reply:
OK, I remembered why I did this: only the scalar u-coordinate of the `NoteCommitment` is ever sent on the wire, not the canonical encoding of the point (v + 255*(u mod 2)), which seems because the full point is never reconstructed for a `NoteCommitment`; the scalar is fed into the prover, and when notes are decrypted, the `NoteCommitment` is reconstructed from scratch and its u-coordinate is compared to the one sent in the clear in the transaction. 

So this would lead me to say, 'a `NoteCommitment` actually wraps `Fq`', but, that's not actually what the spec says, the spec¹ says a `NoteCommitment` is point, like how a `ValueCommitment` is a point. But we can never `deserialize()` or do `NoteCommitment::try_from([u8; 32])` from public info as part of a transaction and get that point, we only get a u-coordinate scalar. 

So, should there be a `NoteCommitmentU` perhaps? Or just a plain `jubjub::Fq` in the transaction structs? Maybe the latter is fine, and then a `NoteCommitment` wraps a `jubjub::AffinePoint` like `ValueCommitment`. I'll go with that for now and we'll see how it feels but open to discussion.

¹![image](https://user-images.githubusercontent.com/552961/89703178-3af71280-d916-11ea-8877-6f65a80df527.png)

Reply:
I wrote that whole note without realizing this is about  Sprout note commitments 🙃🤡

Reply:
`sprout::NoteCommitment`s are just bytes because they are literally just SHA-256 hashes

![image](https://user-images.githubusercontent.com/552961/89703439-af32b580-d918-11ea-83c5-f69413324fcb.png)